src/buf.c

```diff
@@ -93,6 +95,15 @@ void rs_filebuf_free(rs_filebuf_t *fb)
 }
 
 
+void rs_get_filesize(FILE *f, rs_long_t *size)
+{
+    rs_long_t pos = ftell(f);
+    fseek(f, 0, SEEK_END);
```

Comment:
Also, this needs to allow for reading from a pipe where we can't get the length.